ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/question_answering/QuestionAnsweringModel.java

```diff
@@ -28,46 +28,110 @@
         predictor.predict(input);
     }
 
+    /**
+     * Checks and sets the highlighting type based on model configuration.
+     *
+     * @param modelConfig The model configuration
+     */
+    private void checkHighlightingType(MLModelConfig modelConfig) {
+        if (modelConfig != null) {
+            if (modelConfig.getModelType() != null) {
+                String modelType = modelConfig.getModelType();
+                if (SENTENCE_HIGHLIGHTING_TYPE.equalsIgnoreCase(modelType)) {
+                    highlightingType = SENTENCE_HIGHLIGHTING_TYPE;
+                }
+            }
+        }
+    }
+
     @Override
     public ModelTensorOutput predict(String modelId, MLInput mlInput) throws TranslateException {
         MLInputDataset inputDataSet = mlInput.getInputDataset();
-        List<ModelTensors> tensorOutputs = new ArrayList<>();
-        Output output;
         QuestionAnsweringInputDataSet qaInputDataSet = (QuestionAnsweringInputDataSet) inputDataSet;
         String question = qaInputDataSet.getQuestion();
         String context = qaInputDataSet.getContext();
+
+        if (SENTENCE_HIGHLIGHTING_TYPE.equals(highlightingType)) {
+            return predictWithSentenceHighlighting(question, context);
+        }
+        return predictWithStandardQA(question, context);
+    }
+
+    /**
+     * Performs standard question answering without highlighting.
+     *
+     * @param question The question to answer
+     * @param context The context to search for answers
+     * @return Model output with answer
+     * @throws TranslateException if prediction fails
+     */
+    private ModelTensorOutput predictWithStandardQA(String question, String context) throws TranslateException {
+        List<ModelTensors> tensorOutputs = new ArrayList<>();
         Input input = new Input();
         input.add(question);
         input.add(context);
-        output = getPredictor().predict(input);
+        Output output = getPredictor().predict(input);
         tensorOutputs.add(parseModelTensorOutput(output, null));
         return new ModelTensorOutput(tensorOutputs);
     }
 
+    /**
+     * Performs question answering with sentence-level highlighting.
+     *
+     * @param question The question to answer
+     * @param context The context to search for answers
+     * @return Model output with highlighted sentences
+     * @throws TranslateException if prediction fails
+     */
+    private ModelTensorOutput predictWithSentenceHighlighting(String question, String context) throws TranslateException {
+        Input input = new Input();
+        input.add(FIELD_QUESTION, question);
+        input.add(FIELD_CONTEXT, context);
+
+        Output output = getPredictor().predict(input);
+        ModelTensors modelTensors = parseModelTensorOutput(output, null);
+        List<ModelTensors> tensorOutputs = new ArrayList<>();
+        tensorOutputs.add(modelTensors);
+
+        return new ModelTensorOutput(tensorOutputs);
+    }
```

Comment:
I don't see any difference between these two functions. Actually the predict should have the same input and output for general QA. The different thing is the translators, I think.

Reply:
Updated